cai_causal_graph/time_series_causal_graph.py

```diff
@@ -1042,6 +1042,12 @@ def from_adjacency_matrices(
         # create the empty graph
         tsgraph = TimeSeriesCausalGraph()
 
+        # first add all the nodes (there could be floating nodes)
+        lags = list(adjacency_matrices.keys())
+        for variable_name in variable_names_str:
+            for lag in lags:
```

Comment:
Do we definitely want to add all lags? I'm guessing we do but just thinking in terms of minimal graphs. This would mean that CD methods using this would need to be modded to now call .get_minimal_graph after initialising from adjacencies.

Reply:
Oh yeah agree, but I thought you wanted all the nodes actually.

Reply:
I think this is worth clarifying in the docstring for the method so it is documented somewhere that floating nodes at time lag != 0 are not added.

Reply:
Ok agree, adding now!